pkg/k8s/portforward.go

```diff
@@ -162,6 +164,42 @@ func (pf *PortForward) Run() error {
 	return fw.ForwardPorts()
 }
 
+// Init creates and runs a port-forward connection.
+// This function blocks until the connection is established, in which case it returns nil.
+// It's the caller's responsibility to call Stop() to finish the connection.
+func (pf *PortForward) Init() error {
+	log.Debugf("Starting port forward to %s %d:%d", pf.url, pf.localPort, pf.remotePort)
+
+	failure := make(chan error)
+
+	go func() {
+		if err := pf.run(); err != nil {
+			failure <- err
+		}
+
+		select {
+		case <-pf.GetStop():
+			// stopCh was closed, do nothing
+		default:
+			// pf.run() returned for some other reason, close stopCh
+			pf.Stop()
+		}
+	}()
+
+	// The `select` statement below depends on one of two outcomes from `pf.Run()`:
+	// 1) Succeed and block, causing a receive on `<-pf.Ready()`
+	// 2) Return an err, causing a receive `<-failure`
+	select {
+	case <-pf.Ready():
```

Comment:
sorry i missed this on the last review, i think you can just do `<-pf.readyCh` and remove the `Ready()` method.

Reply:
Good :eye: , `Ready()` is not being used anywhere else :+1: